### PR TITLE
Implement labelOrEntityFieldLabel and replace misuses of name prop

### LINKED ISSUE #3892: Use name only as a fallback for the label if the label is null
_Originally posted by @usu in https://github.com/ecamp/ecamp3/pull/3824#discussion_r1341692852_

### PR DESCRIPTION
**Pull request by manuelmeister**
Fixes #3892

Simplify labels and use defaultLabel to use the entity field translations. Also the name prop should not be translated, but should be a constant string.

### CODE REVIEW

File: frontend/src/components/campAdmin/DialogMaterialListForm.vue [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
-  </div>
+  </e-form>
 </template>
 
 <script>
+import EForm from '@/components/form/base/EForm.vue'
+
 export default {
   name: 'DialogMaterialListForm',
+  components: { EForm },
```
Comment (carlobeltrame) [inline frontend/src/components/campAdmin/DialogMaterialListForm.vue]: Should we add EForm to the list of autoloaded components? Apparently all our E-input components are autoloaded, and according to the change you are doing here, pretty much every timewe use an E-input, we also use an EForm

File: frontend/src/components/campAdmin/DialogPeriodDateEdit.vue [hunk 2]
```diff
@@ -12,94 +12,75 @@
...
+        />
       </div>
-    </div>
-    <div v-if="mode == 'changeEnd'" class="e-form-container">
-      <p>{{ $tc('components.campAdmin.dialogPeriodDateEdit.periodChangeEnd') }}</p>
-      <div class="e-form-container">
-        <v-text-field
+      <div v-if="mode === 'changeEnd'" class="e-form-container">
+        <p>{{ $tc('components.campAdmin.dialogPeriodDateEdit.periodChangeEnd') }}</p>
+        <e-text-field
```
Comment (carlobeltrame) [inline frontend/src/components/campAdmin/DialogPeriodDateEdit.vue:50]: Nice catch using an e-textfield instead of a v-textfield here

File: frontend/src/components/campCreate/CampCreateStep2.vue [hunk 3]
```diff
@@ -8,7 +8,8 @@
             v-model="localCamp.campPrototype"
             :vee-rules="{ required: false, excluded: ['', false, true] }"
             :skip-if-empty="false"
-            :name="$tc('entity.camp.prototype')"
+            name="prototype"
+            :label="$tc('entity.camp.prototype')"
```
Comment (carlobeltrame) [inline frontend/src/components/campCreate/CampCreateStep2.vue:12]: Is it not possible to provide entityName or use an EForm here?

File: frontend/src/components/form/base/BaseComponent.vue [hunk 4]
```diff
@@ -7,11 +7,12 @@
   >
     <component
       :is="inputComponent"
+      :name="name"
       v-bind="$attrs"
       :filled="filled"
       :hide-details="hideDetails"
       :error-messages="veeErrors.concat(errorMessages)"
-      :label.prop="label || name"
+      :label.prop="defaultLabel"
```
Comment (carlobeltrame) [inline frontend/src/components/form/base/BaseComponent.vue:15]: This confused me at first, because I thought the label can't be overridden anymore. Could we call this labelOrDefault or similar instead?

File: frontend/src/components/form/base/BaseComponent.vue [hunk 5]
```diff
@@ -28,10 +29,11 @@
 import { ValidationProvider } from 'vee-validate'
 import { formComponentPropsMixin } from '@/mixins/formComponentPropsMixin.js'
 import { VTextField } from 'vuetify/lib'
+import VTiptapEditor from '@/components/form/tiptap/VTiptapEditor.vue'
 
 export default {
   name: 'BaseComponent',
-  components: { ValidationProvider, VTextField },
+  components: { VTiptapEditor, ValidationProvider, VTextField },
```
Comment (carlobeltrame) [inline frontend/src/components/form/base/BaseComponent.vue]: Why this change?

File: frontend/src/mixins/formComponentPropsMixin.js [hunk 6]
```diff
@@ -46,4 +49,15 @@ export const formComponentPropsMixin = {
...
+    defaultLabel() {
+      if (this.label !== undefined) {
+        return this.label
+      }
+      if (!this.entityName || !this.name) {
+        return null
+      }
+      return this.$t(`entity.${this.entityName}.fields.${this.name}`)
+    },
+  },
```
Comment (manuelmeister) [inline frontend/src/mixins/formComponentPropsMixin.js]: DefaultLabel if entityName and field name is set.

File: frontend/src/mixins/formComponentPropsMixin.js [hunk 7]
```diff
@@ -46,4 +49,15 @@ export const formComponentPropsMixin = {
       default: () => [],
     },
   },
+  computed: {
+    defaultLabel() {
+      if (this.label !== undefined) {
+        return this.label
+      }
+      if (!this.entityName || !this.name) {
```
Comment (carlobeltrame) [inline frontend/src/mixins/formComponentPropsMixin.js:57]: Should we output a warning somewhere when this happens?

**Review by carlobeltrame — COMMENTED** [review]
Code lgtm, left some suggestions. Since you touched pretty much every form, we should thoroughly test this PR manually

File: frontend/src/components/campCreate/CampCreateStep2.vue (hunk 3 above)
Comment (manuelmeister) [reply frontend/src/components/campCreate/CampCreateStep2.vue:12]: Currently this is not an entity field name, and I'm not sure if it should be.

File: frontend/src/mixins/formComponentPropsMixin.js (hunk 7 above)
Comment (manuelmeister) [reply frontend/src/mixins/formComponentPropsMixin.js:57]: No, because from my point of view `entityName` and field `name` together are like an opt in into this functionality. If you then don't want a label or override a label you can set it to `null`, `false` or `"Mein Label"`.

File: frontend/src/components/campAdmin/DialogMaterialListForm.vue (hunk 1 above)
Comment (manuelmeister) [reply frontend/src/components/campAdmin/DialogMaterialListForm.vue]: Agreed, fixed in b3b1397

File: frontend/src/components/form/base/BaseComponent.vue (hunk 4 above)
Comment (manuelmeister) [reply frontend/src/components/form/base/BaseComponent.vue:15]: Good point. I now called it `labelOrEntityFieldLabel`

File: frontend/src/components/form/base/BaseComponent.vue (hunk 5 above)
Comment (manuelmeister) [reply frontend/src/components/form/base/BaseComponent.vue]: [Remove automatically added reference by PhpStorm](https://github.com/ecamp/ecamp3/pull/4376/commits/0d9b800425b80a53321d9a831c5c4f701afbb579)

File: frontend/src/components/program/FormScheduleEntryItem.vue [hunk 8]
```diff
@@ -5,7 +5,8 @@
         <e-date-picker
           v-model="localScheduleEntry.start"
           value-format="YYYY-MM-DDTHH:mm:ssZ"
-          :name="$tc('components.program.formScheduleEntryItem.start')"
+          :label="$tc('components.program.formScheduleEntryItem.start')"
+          name="startDate"
```
Comment (BacLuc) [inline frontend/src/components/program/FormScheduleEntryItem.vue:9]: minor:
```suggestion
          name="startDate"
          :label="$tc('components.program.formScheduleEntryItem.start')"
```


I would keep the order in this PR, and specially in this file, consistent

**Comment by manuelmeister** [comment]
## Core Meeting Decision

- rename `fieldname` / `name` to `path` and close #2945
- evt. validation label override prop

**Comment by manuelmeister** [comment]
Superceded by #4746
